be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -152,18 +153,33 @@ Status JsonScanner::_construct_cast_exprs() {
 }
 
 Status JsonScanner::_parse_json_paths(const std::string& jsonpath, std::vector<std::vector<JsonPath>>* path_vecs) {
-    rapidjson::Document doc;
-    doc.Parse(jsonpath.c_str(), jsonpath.length());
-    if (doc.HasParseError() || !doc.IsArray()) {
+    simdjson::dom::parser parser;
+    simdjson::dom::element elem;
+
+    auto err = parser.parse(jsonpath.c_str(), jsonpath.length()).get(elem);
+
+    if (err || !elem.is_array()) {
+        return Status::InvalidArgument(strings::Substitute("Invalid json path: $0", jsonpath));
+    }
+
+    simdjson::dom::array paths;
+    err = elem.get_array().get(paths);
+    if (err) {
         return Status::InvalidArgument(strings::Substitute("Invalid json path: $0", jsonpath));
     }
-    for (int i = 0; i < doc.Size(); i++) {
-        const rapidjson::Value& path = doc[i];
-        if (!path.IsString()) {
+
+    for (auto path : paths) {
+        if (!path.is_string()) {
             return Status::InvalidArgument(strings::Substitute("Invalid json path: $0", jsonpath));
         }
         std::vector<JsonPath> parsed_paths;
-        JsonFunctions::parse_json_paths(path.GetString(), &parsed_paths);
+        const char* cstr;
+        auto err = path.get_c_str().get(cstr);
+        if (err) {
+            return Status::InvalidArgument(strings::Substitute("Invalid json path: $0", jsonpath));
+        }
+
+        JsonFunctions::parse_json_paths(std::string(cstr), &parsed_paths);
```

Comment:
Can we eliminate the `std::string` construction?

Reply:
There's no easy way to get a `std::string` from `path`, which is in type `simdjson::dom::element`.